src/containers/common/ProgramHomeContainer.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Actions } from 'jumpstate';
+import { Switch, Route, Redirect } from 'react-router-dom';
+
+import AstroHome from '../../components/astro/AstroHome';
+import DarienRoutesContainer from '../darien/DarienRoutesContainer';
+
+import {
+  PROGRAMS_INITIAL_STATE, PROGRAMS_PROPTYPES
+} from '../../ducks/programs';
+
+export class ProgramHomeContainer extends React.Component {
+  constructor() {
+    super();
+  }
+
+  componentDidMount() {
+    if (!this.props.selectedProgram) {
+      Actions.getProgram({ programs: this.props.programs, param: this.props.match.params.program });
+    }
+  }
+
+  componentWillUnmount() {
+    Actions.programs.selectProgram(PROGRAMS_INITIAL_STATE.program);
+  }
+
+  render() {
+    // How do I pass props down consistently? They only get initial props, not the updated ones.
```

Comment:
What were you trying to pass down, but didn't get through/the update didn't register? The props.selectedProgram?